examples/voice-search/voiceSearchPlugin.tsx

```diff
@@ -0,0 +1,167 @@
+/** @jsx h */
+import { AutocompletePlugin } from '@algolia/autocomplete-js';
+import { h, render } from 'preact';
+
+import { createVoiceSearch, VoiceSearchStatus } from './voiceSearchApi';
+
+type CreateVoiceSearchPluginParams = {
+  /**
+   * Sets the language of the speech recognition.
+   *
+   * If not specified, this defaults to the HTML `lang` attribute value, or the user agent's language setting if that isn't set either.
+   *
+   * @example "en-US"
+   */
+  language?: string;
+};
+
+export function createVoiceSearchPlugin({
+  language,
+}: CreateVoiceSearchPluginParams): AutocompletePlugin<any, undefined> {
+  return {
+    subscribe({ setQuery, setIsOpen, refresh }) {
+      requestAnimationFrame(() => {
```

Comment:
This confirms we need to:
- be able to listen to Detached Mode events
- access the searchbox rendering (even though we can circumvent it with the DOM API)